src/main/scala/com/github/matsluni/akkahttpspi/AkkaHttpClient.scala

```diff
@@ -145,21 +145,31 @@ object AkkaHttpClient {
       implicit val ec = executionContext.getOrElse(as.dispatcher)
       val mat: Materializer = SystemMaterializer(as).materializer
 
+      val mergedAttributeMap = attributeMap.merge(SdkHttpConfigurationOption.GLOBAL_HTTP_DEFAULTS)
```

Comment:
Except from `Protocol` nothing else is used from the `GLOBAL_HTTP_DEFAULTS`. Should we create a attributeMap with just the `Protocol` in it, with a default to Http/1.1?

Otherwise it may be confusing, what other settings are taken from there and right now I would prefer to configure the Akka-Http client via its normal methods of configuration.

WDYT?

Reply:
> Except from Protocol nothing else is used from the GLOBAL_HTTP_DEFAULTS

The use of `GLOBAL_HTTP_DEFAULTS`, was inspired in the [NettyNioAsyncHttpClient](https://github.com/aws/aws-sdk-java-v2/blob/13985e0668a9a0b12ad331644e3c4fd1385c2cd7/http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.java#L741-L752), and I was planning to submit another PR that reads other attributes from there like `CONNECTION_TIMEOUT`, `MAX_CONNECTIONS`, etc...

> I would prefer to configure the Akka-Http client via its normal methods of configuration.

I understand right now that configuration is done via the normal methods of configuration i.e typesafe config library, but what are your thoughts about changing that to use the attributeMap? Asking because I see some specific clients configuring specific options like [KinesisHttpConfigurationOptions](https://github.com/aws/aws-sdk-java-v2/blob/13985e0668a9a0b12ad331644e3c4fd1385c2cd7/services/kinesis/src/main/java/software/amazon/awssdk/services/kinesis/internal/KinesisHttpConfigurationOptions.java#L27C45-L32) and you can find other examples [here](https://github.com/search?q=repo%3Aaws%2Faws-sdk-java-v2+SdkHttpConfigurationOption+path%3A%2F%5Eservices%5C%2F%2F&type=code)
Maybe we should honor those...

Reply:
I removed the use `GLOBAL_HTTP_DEFAULTS` and `SdkHttpConfigurationOption` and added a simple protocol field to the builder

Reply:
I created the issue https://github.com/matsluni/aws-spi-akka-http/issues/232 to track your idea with the config changes.